### PR TITLE
Sarathbrp patch 3

### PR DESCRIPTION
**Pull request by sarathbrp**
assert logic replaced with if condition with a warning

### CODE REVIEW

File: opl/cluster_read.py [hunk 1]
```diff
@@ -88,18 +88,24 @@ def measure(self, start, end, name, monitoring_query, monitoring_step):
...
-            "'data' needs to be in response"
-        assert 'result' in json_response['data'], \
-            "'result' needs to be in response's 'data'"
-        assert len(json_response['data']['result']) != 0, \
-            "missing 'response' in response's 'data'"
-        assert len(json_response['data']['result']) == 1, \
-            "we need exactly one 'response' in response's 'data'"
-        assert 'values' in json_response['data']['result'][0], \
-            "we need expected form of response"
+
```
Comment (jhutar) [inline opl/cluster_read.py]: Do not remove these asserts. You whould have to remove them in other plugins as well. Just replace that `traceback.print_exc()` on line ~312 with something that would log that traceback as a debug and print a warning there.

File: opl/hbi_utils.py [hunk 2]
```diff
@@ -105,46 +105,43 @@ def wait_for_next_second():
             f"Not all messages sent {data_stats['successes']} + {data_stats['failures']} != {args.count}"
         )
 
+        
+def fetch_records_count(inventory):
```
Comment (jhutar) [inline opl/hbi_utils.py:109]: Please do not mix this cluster_read change with hbi_utils change in one PR.

**Review by jhutar — CHANGES_REQUESTED** [review]
Se the comments. Please also note I think your patch needs rebase.

**Comment by jhutar** [comment]
Please do not mix this cluster_read change with hbi_utils change in one PR. Or are they related somehow?

File: opl/cluster_read.py [hunk 3]
```diff
@@ -309,7 +310,8 @@ def __next__(self):
                     return instance.measure(self.start, self.end, **self.config[i])
                 except Exception as e:
                     logging.error(f"Failed to measure {self.config[i]['name']}: {e}")
-                    traceback.print_exc()
+                    tb = traceback.format_exc()
+                    print(tb, file=sys.stderr)
```
Comment (jhutar) [inline opl/cluster_read.py]: IMO you can just use https://docs.python.org/3/library/logging.html#logging.exception

File: opl/hbi_utils.py (hunk 2 above)
Comment (sarathbrp) [reply opl/hbi_utils.py:109]: This code exists in master branch no changes required in this merge request.

File: opl/cluster_read.py (hunk 3 above)
Comment (sarathbrp) [reply opl/cluster_read.py]: Done. Replaced logging.error with logging.exception
logging.exception will log the exception message and trace back.

File: opl/hbi_utils.py (hunk 2 above)
Comment (jhutar) [reply opl/hbi_utils.py:109]: So it should not be in the PR. Please rebase your PR.

File: opl/cluster_read.py (hunk 3 above)
Comment (jhutar) [reply opl/cluster_read.py]: Probably you forgot to push the changes.

**Pull request closed without merging**